### DISCUSSION TITLE
How to use generated atlas? png+json?

### DISCUSSION THREAD

**Discussion by sergeyvin** [Chlumsky/msdf-atlas-gen#30] [Q&A] [answered]
Could you add sample code how to use generated atlas?  Do we need to use both PNG file for texture and JSON/CSV file for glyph offsets ?  
Sample fragment shader code was given in msdf  rep description https://github.com/Chlumsky/msdfgen . Thanks !

**Comment by Chlumsky** [comment, marked as answer]
Yes, how else would you know where the glyphs are? Here is code that generates vertex data for a given text: https://github.com/Chlumsky/msdf-atlas-gen/blob/dadc65c85c5da88b573c5278174a1e0df71914b4/msdf-atlas-gen/shadron-preview-generator.cpp#L107-L143

The layout of each vertex is: https://github.com/Chlumsky/msdf-atlas-gen/blob/dadc65c85c5da88b573c5278174a1e0df71914b4/msdf-atlas-gen/shadron-preview-generator.cpp#L30-L33
